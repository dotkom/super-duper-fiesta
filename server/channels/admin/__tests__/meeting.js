jest.mock('../../../utils');
jest.mock('../../../models/meeting');
const { broadcast, emit } = require('../../../utils');
const { toggleRegistration } = require('../meeting');
const { getActiveGenfors, toggleRegistrationStatus } = require('../../../models/meeting');
const { generateSocket, generateGenfors } = require('../../../utils/generateTestData');

describe('toggleRegistration', () => {
  beforeEach(() => {
    getActiveGenfors.mockImplementation(async () => generateGenfors());
    toggleRegistrationStatus.mockImplementation(
      async (genfors, registrationOpen) =>
        generateGenfors({ ...genfors, registrationOpen: !registrationOpen }));
  });

  const generateData = data => (Object.assign({
    registrationOpen: true,
  }, data));

  it('emits a toggle action that closes registration', async () => {
    await toggleRegistration(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });

  it('emits a toggle action that opens registration', async () => {
    await toggleRegistration(generateSocket(), generateData({ registrationOpen: false }));

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });
});
