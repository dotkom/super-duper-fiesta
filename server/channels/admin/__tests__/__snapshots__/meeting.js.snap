// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleRegistration emits a toggle action that closes registration 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": false,
          "genfors": "1",
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "TOGGLED_REGISTRATION_STATE",
    Object {
      "id": 1,
      "pin": 3141592653,
      "registrationOpen": false,
    },
  ],
]
`;

exports[`toggleRegistration emits a toggle action that opens registration 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": false,
          "genfors": "1",
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "TOGGLED_REGISTRATION_STATE",
    Object {
      "id": 1,
      "pin": 3141592653,
      "registrationOpen": true,
    },
  ],
]
`;
